docs/en/query_language/functions/other_functions.md

```diff
@@ -178,6 +178,39 @@ Accepts Float32 and Float64 and returns UInt8 equal to 1 if the argument is not
 
 Accepts Float32 and Float64 and returns UInt8 equal to 1 if the argument is infinite, otherwise 0. Note that 0 is returned for a NaN.
 
+## ifNotFinite {#ifnotfinite}
+
+Accepts Float32 and Float64. The function has the [ternary operator](https://clickhouse.tech/docs/en/query_language/functions/conditional_functions/#ternary-operator) structure: `ifNotFinite(x, y) = isFinite(x) ? x : y`. It returns `x` if `x` if finite, and `y` if `x` is not finite. 
+
+**Syntax** 
+
+```
+ifNotFinite(x,y)
+```
+**Parameters** 
+
+- `x` — Float32 or Float64
+- `y` — Float32 or Float64
+
+**Returned value**
+
+- returns `x` if `x` if finite, and `y` if `x` is not finite
+
+**Example**
+
+Query:
+
+```
+SELECT ifNotFinite(sqrt(-1), -42)
```

Comment:
sqrt(-1) is not Infimum, it is NaN. 

This example `select 1/0 as infimum, ifNotFinite(infimum,42)` is more correct.